xclim/sdba/properties.py

```diff
@@ -0,0 +1,737 @@
+    Alavoine, M., & Grenier, P. (2021). The distinct problems of physical inconsistency and of multivariate bias
+    potentially involved in the statistical adjustment of climate simulations. California Digital Library (CDL).
+    https://doi.org/10.31223/x5c34c
+
+    Examples
+    --------
+    >>> from xclim.testing import open_dataset
+    >>> pr = open_dataset(path_to_pr_file).pr
+    >>> acf(da=pr, lag=3, time_res='season')
+    """
+    if time_res == "year":
+        raise ValueError(
+            "'year' is not a valid time resolution for this statistical property."
+        )
+
+    attrs = da.attrs
+    da = da.resample(time=res2freq[time_res])
+
+    def acf_last(x, nlags):
+        """statsmodels acf calculates acf for lag 0 to nlags, this return only the last one."""
+        out_last = stattools.acf(x, nlags=nlags)
+        return out_last[-1]
+
+    out = xr.apply_ufunc(
+        acf_last,
+        da,
+        input_core_dims=[["time"]],
+        vectorize=True,
+        kwargs={"nlags": lag},
+        dask="parallelized",
+    )
+    # average over the years
+    out = out.groupby(f"__resample_dim__.{time_res}").mean(dim="__resample_dim__")
+
+    out.attrs.update(attrs)
+    out.attrs["long_name"] = f"lag-{lag} autocorrelation of {attrs['long_name']}"
+    out.attrs["units"] = ""
+    return out
+
+
+# time_res was kept even though 'year' it the only acceptable arg to keep the signature similar to other properties
+@update_xclim_history
+@register_statistical_properties(aspect="temporal", seasonal=False, annual=True)
+def annual_cycle_amplitude(
+    da: xr.DataArray, amplitude_type: str = "absolute", time_res: str = "year"
+) -> xr.DataArray:
+    r"""Annual Cycle amplitude.
+
+    The amplitudes of the annual cycle are calculated for each year, than averaged over the all years.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Variable on which to calculate the diagnostic.
+
+    amplitude_type: {'absolute','relative'}
+        Type of amplitude.
+        'absolute' is the peak-to-peak amplitude. (max - min)
+        'relative' is a relative percentage. 100 * (max - min) / mean. Recommanded for precipitation.
+
+    Returns
+    -------
+    out: xr.DataArray,
+      {amplitude_type} amplitude of the annual cycle.
+
+    Examples
+    --------
+    >>> from xclim.testing import open_dataset
+    >>> pr = open_dataset(path_to_pr_file).pr
+    >>> annual_cycle_amplitude(da=pr, amplitude_type='relative')
+    """
+    if time_res != "year":
+        raise ValueError(
+            "'year' is the only valid time resolution for this statistical property."
+        )
+
+    attrs = da.attrs
+    da = da.resample(time="YS")
+    # amplitude
+    amp = da.max(dim="time") - da.min(dim="time")
+    amp.attrs.update(attrs)
+    if (
+        xc.core.units.units2pint(attrs["units"]).dimensionality
+        == xc.core.units.units2pint("degC").dimensionality
+    ):
+        amp.attrs["units"] = "delta_degree_Celsius"
```

Comment:
What if `da` is in °F ?

I'd suggest this:
```suggestion
    d = 1 * xc.core.units.units2pint(attrs["units"])
    amp.attrs["units"] = xc.core.units.pint2cfunits(d - d)
```

But I now see that this gives `°K` with Kelvins, so doesn't achieve your goal. 
Hum. We might be better off with an helper function in `units.py`, I'll open an issue.

Reply:
oups I forgot about fahrenheit!

Reply:
I just added `ensure_delta` in `units.py`. I didn't put `has_dimensionality(units, dim)` as you ask in issue #968, because `ensure_delta` doesn't need to check that. For units that don't have a delta (all but temperature I think), it just returns the unit as is.